samza-sql/src/main/java/org/apache/samza/sql/interfaces/SourceResolver.java

```diff
@@ -0,0 +1,34 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.interfaces;
+
+/**
+ * Source Resolvers are used by Samza Sql application to fetch the {@link SystemStreamConfig} corresponding to the source.
+ */
+public interface SourceResolver {
```

Comment:
Is this just a config builder? SourceResolver should return some kind of Source<V> I assume.

Reply:
I am not sure i understood the question. The API returns the Source Which is SystemStream in the samza's nomenclature. Are you saying i should change the name to SystemStreamResolver?